### ISSUE TITLE
Unable to install  json bundler dependency on OS/X 

### ISSUE THREAD

**Issue by javadba** [vega/vega-lite#2364]
TL;DR we get error  "file was built for x86_64 which is not the architecture being linked (i386):".

Details:

The bundler depends on json. Trying to install the latter results in:

sudo gem install json -v '1.8.6'
Building native extensions.  This could take a while...
ERROR:  Error installing json:
	ERROR: Failed to build gem native extension.

    /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby extconf.rb
creating Makefile

make "DESTDIR="
compiling generator.c
linking shared-object json/ext/generator.bundle
ld: in '/usr/local/lib/libunwind.dylib', file was built for x86_64 which is not the architecture being linked (i386): /usr/local/lib/libunwind.dylib for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [generator.bundle] Error 1

Gem files will remain installed in /Library/Ruby/Gems/2.0.0/gems/json-1.8.6 for inspection.
Results logged to /Library/Ruby/Gems/2.0.0/gems/json-1.8.6/ext/json/ext/generator/gem_make.out

**Comment by kanitw** [comment]
Can you provide *detailed* commands that you run to produce these errors?  Without that we won't be able to help.    (I'm not sure why you need ruby bundler)

**Comment by javadba** [comment]
Here it is


    bundle install
    Fetching gem metadata from https://rubygems.org/..............
    Fetching version metadata from https://rubygems.org/...
    Fetching dependency metadata from https://rubygems.org/..
    Resolving dependencies...
    ..

    Installing mini_portile2 2.1.0
    Installing jekyll-paginate 1.1.0
    Installing jekyll-swiss 0.4.0
    Installing unicode-display_width 1.1.3
    Using bundler 1.14.6
    Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.

        /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby extconf.rb
    creating Makefile

    make "DESTDIR="
    compiling generator.c
    linking shared-object json/ext/generator.bundle
    ld: in '/usr/local/lib/libunwind.dylib', file was built for x86_64 which is not the architecture being linked (i386): /usr/local/lib/libunwind.dylib for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make: *** [generator.bundle] Error 1


    Gem files will remain installed in /var/folders/0g/w9k58vxn03nbrwt220lxjr6c1zj68n/T/bundler20170514-48401-hdzym2json-1.8.6/gems/json-1.8.6 for inspection.
    Results logged to /var/folders/0g/w9k58vxn03nbrwt220lxjr6c1zj68n/T/bundler20170514-48401-hdzym2json-1.8.6/gems/json-1.8.6/ext/json/ext/generator/gem_make.out

    An error occurred while installing json (1.8.6), and Bundler cannot continue.
    Make sure that `gem install json -v '1.8.6'` succeeds before bundling.

So *I attempted what the error message asked* 

    sudo gem install json -v '1.8.6'

    Building native extensions.  This could take a while...
    ERROR:  Error installing json:
      ERROR: Failed to build gem native extension.

        /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby extconf.rb
    creating Makefile

    make "DESTDIR="
    compiling generator.c
    linking shared-object json/ext/generator.bundle
    ld: in '/usr/local/lib/libunwind.dylib', file was built for x86_64 which is not the architecture being linked (i386): /usr/local/lib/libunwind.dylib for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make: *** [generator.bundle] Error 1


But when I *re-ran bundler* it got further anyways 


bundler install 

Gem::InstallError: nokogiri requires Ruby version >= 2.1.0.
An error occurred while installing nokogiri (1.7.2), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.7.2'` succeeds before bundling.

    brew install ruby
    sudo gem install nokogiri -v '1.7.2'

    sudo gem install nokogiri -v '1.7.2'
    Building native extensions.  This could take a while...
    Successfully installed nokogiri-1.7.2
    Parsing documentation for nokogiri-1.7.2
    Done installing documentation for nokogiri after 1 seconds
    1 gem installed


    Installing nokogiri 1.6.8.1 with native extensions
    Using terminal-table 1.7.3
    Using activesupport 4.2.7
    Using jekyll-coffeescript 1.0.1
    Using typhoeus 0.8.0
    Using listen 3.0.6
    Using sawyer 0.8.1
    Installing html-pipeline 2.5.0
    Installing jekyll-watch 1.5.0
    Installing octokit 4.7.0
    Installing jekyll 3.4.3
    Installing github-pages-health-check 1.3.3
    Installing jekyll-gist 1.4.0
    Installing jekyll-avatar 0.4.2
    Installing jekyll-default-layout 0.1.4
    Installing jekyll-feed 0.9.2
    Installing jekyll-github-metadata 2.3.1
    Installing jekyll-mentions 1.2.0
    Installing jekyll-optional-front-matter 0.1.2
    Installing jekyll-readme-index 0.1.0
    Installing jekyll-redirect-from 0.12.1
    Installing jekyll-relative-links 0.4.0
    Installing jekyll-seo-tag 2.2.2
    Installing jekyll-sitemap 1.0.0
    Installing jekyll-theme-architect 0.0.4
    Installing jekyll-theme-cayman 0.0.4
    Installing jekyll-theme-dinky 0.0.4
    Installing jekyll-theme-hacker 0.0.4
    Installing jekyll-theme-leap-day 0.0.4
    Installing jekyll-theme-merlot 0.0.4
    Installing jekyll-theme-midnight 0.0.4
    Installing jekyll-theme-minimal 0.0.4
    Installing jekyll-theme-modernist 0.0.4
    Installing jekyll-theme-primer 0.1.8
    Installing jekyll-theme-slate 0.0.4
    Installing jekyll-theme-tactile 0.0.4
    Installing jekyll-theme-time-machine 0.0.4
    Installing jekyll-titles-from-headings 0.1.5
    Installing jemoji 0.8.0
    Installing minima 2.1.1
    Installing github-pages 136
    Bundle complete! 1 Gemfile dependency, 74 gems now installed.
    Use `bundle show [gemname]` to see where a bundled gem is installed.
    Post-install message from html-pipeline:
    -------------------------------------------------
    Thank you for installing html-pipeline!
    You must bundle Filter gem dependencies.
    See html-pipeline README.md for more details.
    https://github.com/jch/html-pipeline#dependencies

Then do `npm run site` which fails big time.

npm run site

    20:28:23/vega-lite $npm run site

    > vega-lite@1.3.1 presite /xfer/vega-lite
    > tsc && npm run build && bower install && npm run data && npm run build:toc

    test/compile/axis.test.ts(149,7): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
      Type argument candidate 'AxisOrient' is not a valid type argument because it is not a supertype of candidate '"bottom"'.
    test/compile/axis.test.ts(170,7): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
      Type argument candidate 'AxisOrient' is not a valid type argument because it is not a supertype of candidate '"top"'.
    test/compile/data.test.ts(285,7): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
      Type argument candidate 'Dict<boolean>' is not a valid type argument because it is not a supertype of candidate '{ qq: { field: string; type: string; }; tt: { field: string; type: string; }; oo: null; }'.
        Property 'qq' is incompatible with index signature.
          Type '{ field: string; type: string; }' is not assignable to type 'boolean'.
    test/compile/data.test.ts(298,7): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
      Type argument candidate 'Dict<boolean>' is not a valid type argument because it is not a supertype of candidate '{ qq: { field: string; type: string; }; tt: { field: string; type: string; }; oo: { field: string...'.
        ..

            npm ERR! Darwin 15.6.0
    npm ERR! argv "/usr/local/Cellar/node/7.10.0/bin/node" "/usr/local/bin/npm" "run" "site"
    npm ERR! node v7.10.0
    npm ERR! npm  v4.2.0
    npm ERR! code ELIFECYCLE
    npm ERR! errno 2
    npm ERR! vega-lite@1.3.1 presite: `tsc && npm run build && bower install && npm run data && npm run build:toc`
    npm ERR! Exit status 2
    npm ERR!
    npm ERR! Failed at the vega-lite@1.3.1 presite script 'tsc && npm run build && bower install && npm run data && npm run build:toc'.
    npm ERR! Make sure you have the latest version of node.js and npm installed.
    npm ERR! If you do, this is most likely a problem with the vega-lite package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     tsc && npm run build && bower install && npm run data && npm run build:toc
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs vega-lite
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!     npm owner ls vega-lite
    npm ERR! There is likely additional logging output above.

    npm ERR! Please include the following file with any support request:

**Comment by NeelMohapatra** [comment]
Did you run `gem install bundler` before running `bundle install`?

**Comment by javadba** [comment]
It is in the list of commands I ran:  `sudo` gem install bundler


2017-05-15 8:46 GMT-07:00 Swojit Mohapatra <notifications@github.com>:

**Comment by domoritz** [comment]
I don't know how to fix the bundler issue you have. 

I'm using `ruby 2.3.4p301 (2017-03-30 revision 58214) [x86_64-darwin16]` and `Bundler version 1.14.6` and things work fine. It also works on travis (ubuntu) with whatever versions they are using. 

`npm run site` seems to fail with version 1.3.1 for you. Can you switch to version 2?

**Comment by domoritz** [comment]
This is an issue with bundler and ruby, not Vega-Lite.
